### PR TITLE
Mutating webhook for injecting secrets into Kubernetes containers

### PR DESCRIPTION
**Pull request by bonifaido**

### CODE REVIEW

File: cmd/vault-secrets-webhook/main.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+	annotations := obj.GetAnnotations()
+	if _, ok := annotations["vault.banzaicloud.com/mutated"]; ok {
+		return false, nil
+	}
+	if annotations == nil {
+		annotations = map[string]string{}
+		obj.SetAnnotations(annotations)
+	}
+	annotations["vault.banzaicloud.com/mutated"] = "true"
```
Comment (stoader) [inline cmd/vault-secrets-webhook/main.go]: should this be rather `vault.banzaicloud.io` or coming from a config?

Comment (bonifaido) [reply cmd/vault-secrets-webhook/main.go]: I think the annotations keys should be fix. We were using vault.banzaicloud.com for the operator, but maybe we should support both!

Comment (matyix) [reply cmd/vault-secrets-webhook/main.go]: In general, we use .io but there might be other changes as well to be in line with other annotations we use (as an example see the pattern used in the [HPA operator](https://github.com/banzaicloud/hpa-operator))

Comment (bonifaido) [reply cmd/vault-secrets-webhook/main.go]: Review comments resolved around annotations.

**Pull request merged**